### ISSUE TITLE
`projectile-find-file` behavior deteriorated in combination with `vertico-prescient` as of fb78b79

### ISSUE THREAD

**Issue by johannes-mueller** [bbatsov/projectile#1902]
This issue has been introduced by fb78b79

## Expected behavior

In a project with the following tree

```
.
├── foo.txt
└── src
    └── foobar.c
```

I would expect `projectile-find-file` to complete from "f" to "foo.txt" and "src/foobar.c"

## Actual behavior

Only "foo.txt" is suggested for completion.

Reverting fb78b79 seems to fix it.

## Steps to reproduce the problem

* Setup a repo with the file tree as above

* Launch emacs with the following init file in the repo dir

```elisp
(require 'package)

(use-package projectile
  :init
  (setq projectile-completion-system 'default)
)

(use-package vertico
  :ensure t
  :init (vertico-mode))

(use-package vertico-prescient
  :init (vertico-prescient-mode))
```

* call `projectile-find-file` and hit 'F', <TAB>

### Projectile version information

```
Projectile 20240824.1029
```

### Emacs version

GNU Emacs 29.4

### Operating system

Manjaro Linux

**Comment by bbatsov** [comment]
@juergenhoetzel Can you take a look at this bug report?

**Comment by juergenhoetzel** [comment]
> @juergenhoetzel Can you take a look at this bug report?

Sorry for the inconvenience. This is fixed by #1903

**Comment by johannes-mueller** [comment]
Confirming the fix. Thanks.
